torch/_inductor/codegen/common.py

```diff
@@ -1747,13 +1754,216 @@ def generate(
 
         return var
 
-    def newvar(self, bounds: ValueRanges[Any] = ValueRanges.unknown()) -> CSEVariable:
+    def newvar(
+        self,
+        dtype: Optional[torch.dtype] = None,
+        bounds: ValueRanges[Any] = ValueRanges.unknown(),
+    ) -> CSEVariable:
         var_name = f"{self.name_prefix}{next(self.iter_buffer_ids)}"
-        var = V.kernel.create_cse_var(var_name, bounds)
+        var = V.kernel.create_cse_var(var_name, bounds, dtype)
         self.varname_map[var_name] = var
         return var
 
 
+def get_promoted_dtype(*args, type_promotion_kind: ELEMENTWISE_TYPE_PROMOTION_KIND):
+    def construct_input(inp):
+        if isinstance(inp, torch._prims_common.Number):
+            return inp
+        else:
+            assert hasattr(inp, "dtype")
+
+            # construct a tmp tensor to use dtype promotion util function
+            return torch.empty(0, dtype=inp.dtype)  # torch.empty(0) has ndim == 1
```

Comment:
`torch.empty(0, dtype=inp.dtype)` actually has different type promotion rules than a tensor because it acts as a scalar. Can you do `torch.empty([1], dtype=inp.dtype)` instead ? Read the docs on type promotion here - https://pytorch.org/docs/stable/tensor_attributes.html#type-promotion-doc

` # torch.empty(0) has ndim == 1` delete this comment  ?